src/Map/src/Map.php

```diff
@@ -86,23 +86,99 @@ public function hasOptions(): bool
         return null !== $this->options;
     }
 
+
+    public function getMarker(string $identifier): ?Marker
+    {
+        foreach ($this->markers as $marker) {
+            if ($this->markers->offsetGet($marker) === $identifier) {
+                return $marker;
+            }
+        }
+
+        return null;
+    }
+
     public function addMarker(Marker $marker): self
     {
-        $this->markers[] = $marker;
+        $this->markers->attach($marker, $marker->identifier ?? $this->markers->getHash($marker));
+
+
+        return $this;
+    }
+
+    public function removeMarker(?Marker $marker): self
+    {
+        if ($marker === null) {
+            return $this;
+        }
+
+        if ($this->markers->contains($marker)) {
+            $this->markers->detach($marker);
+        }
 
         return $this;
     }
```

Comment:
To apply to similar methods too:
```suggestion
    public function removeMarker(Marker|string $markerOrId): self
    {
        $this->markers->remove($markerOrId);

        return $this;
    }
```